operator/internal/manifests/indexgateway.go

```diff
@@ -157,16 +163,19 @@ func NewIndexGatewayStatefulSet(opts Options) *appsv1.StatefulSet {
 
 // NewIndexGatewayGRPCService creates a k8s service for the index-gateway GRPC endpoint
 func NewIndexGatewayGRPCService(opts Options) *corev1.Service {
+	s := serviceNameIndexGatewayGRPC(opts.Name)
 	l := ComponentLabels(LabelIndexGatewayComponent, opts.Name)
+	a := serviceAnnotations(s, opts.Flags.EnableCertificateSigningService)
 
 	return &corev1.Service{
 		TypeMeta: metav1.TypeMeta{
 			Kind:       "Service",
 			APIVersion: corev1.SchemeGroupVersion.String(),
 		},
 		ObjectMeta: metav1.ObjectMeta{
-			Name:   serviceNameIndexGatewayGRPC(opts.Name),
-			Labels: l,
+			Name:        serviceNameIndexGatewayGRPC(opts.Name),
+			Labels:      l,
+			Annotations: a,
```

Comment:
Should we inline `l` and `a` and just keep `s` (maybe renamed to `serviceName`) because it is needed twice?